### PR TITLE
[confighttp] Deprecate HTTPClientSettings, use HTTPClientConfig instead

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Deprecate HTTPClientSettings, use HTTPClientConfig instead

**Link to tracking Issue:**
#6767

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

File: config/confighttp/confighttp.go [hunk 1]
```diff
@@ -103,19 +107,28 @@ type HTTPClientSettings struct {
...
-	return HTTPClientSettings{
+	return HTTPClientConfig{
 		MaxIdleConns:    &maxIdleConns,
 		IdleConnTimeout: &idleConnTimeout,
 	}
 }
 
 // ToClient creates an HTTP client.
-func (hcs *HTTPClientSettings) ToClient(host component.Host, settings component.TelemetrySettings) (*http.Client, error) {
+func (hcs *HTTPClientConfig) ToClient(host component.Host, settings component.TelemetrySettings) (*http.Client, error) {
```
[the remark this answers is not in the document]
Comment (dmitryax) [reply config/confighttp/confighttp.go:131]: Misread it. nvm

File: exporter/otlphttpexporter/config.go [hunk 2]
```diff
@@ -14,9 +14,9 @@ import (
 
 // Config defines configuration for OTLP/HTTP exporter.
 type Config struct {
-	confighttp.HTTPClientSettings `mapstructure:",squash"`     // squash ensures fields are correctly decoded in embedded struct.
-	QueueConfig                   exporterhelper.QueueSettings `mapstructure:"sending_queue"`
-	RetryConfig                   configretry.BackOffConfig    `mapstructure:"retry_on_failure"`
+	confighttp.HTTPClientConfig `mapstructure:",squash"`     // squash ensures fields are correctly decoded in embedded struct.
```
Comment (dmitryax) [inline exporter/otlphttpexporter/config.go:17]: This is a breaking change. the old field should be kept as deprecated

Comment (dmitryax) [reply exporter/otlphttpexporter/config.go:17]: Actually. It'll be pretty hard to keep it backward-compatible. I think we should allow breaking change here. It's only applicable to the API, not the user interface tho. Please mark it accordingly in the changelog

Comment (atoulme) [reply exporter/otlphttpexporter/config.go:17]: yes, I was looking for a way to do struct field aliasing or something like that, but no good solution seems to exist. This only breaks for programmatic use so mostly touching on tests in contrib. I will add to changelog, thanks!

**Pull request merged**